subworkflows/local/generate_html_report/main.nf

```diff
@@ -0,0 +1,211 @@
+import groovy.json.JsonOutput
+import groovy.json.JsonSlurper
+
+
+workflow GENERATE_HTML_REPORT_WORKFLOW {
+    take:
+    barcode_results            // channel: [ val(meta), [ barcode_results ] ]
+    fcs_adaptor
+    trim_ns_results            // channel: [ val(meta), [ trim_ns_results ] ]
+    vecscreen_results          // channel: [ val(meta), [ vecscreen_results ] ]
+    autofilter_results         // channel: [ val(meta), [ autofilter_results ] ]
+    merged_table               // channel: [ val(meta), [ merged_table ] ]
+    phylum_counts              // channel: [ val(meta), [ phylum_counts ] ] - New input for phylum coverage data
+    kmers_results              // channel: [ val(meta), [ kmers_results_dirs ] ]
+    reference_fasta            // channel: [ val(meta), [ reference_fasta ] ]
+    fasta_sanitation_log       // channel: [ val(meta), [ fasta_sanitation_log ] ]
+    fasta_length_filtering_log // channel: [ val(meta), [ fasta_length_filtering_log ] ]
+    jinja_templates_list       // channel: [ jinja_templates_list ] // Updated to accept a list
+    samplesheet                // channel: [ samplesheet ]
+    params_file                // channel: [ params_file ]
+    fcsgx_report_txt           // channel: [ val(meta), [ fcsgx_report_txt ] ]
+    fcsgx_taxonomy_rpt         // channel: [ val(meta), [ fcsgx_taxonomy_rpt ] ]
+    btk_dataset                // channel: [ val(meta), [ btk_dataset ] ]
+    css_files_list             // channel: [ css_files_list ] // CSS files to include in the report
+
+    main:
+    ch_versions = Channel.empty()
+
+    // Convert params to JSON for passing to the HTML report
+    def paramsJson = JsonOutput.toJson(params)
+
+    //
+    // LOGIC: COMBINE ALL INPUT CHANNELS AND ANNOTATE THEM WITH PROCESS TAGS
+    //
+    fcs_adaptor
+        .multiMap { meta, file1, file2 ->
+            euk:    [meta + [process: "FCS_ADAPTOR_EUK"],   file1]
+            prok:   [meta + [process: "FCS_ADAPTOR_PROK"],  file2]
+        }
+        .set { fcs_adaptor_split }
+
+    barcode_results
+        .map{ meta, _file ->
+            def new_meta = meta + [process: "BARCODES"]
+            [new_meta, _file]
+        }.mix(
+            trim_ns_results
+                .map{ meta, _file ->
+                    def new_meta = meta + [process: "TRAILING_NS"]
+                    [new_meta, _file]
+                },
+            autofilter_results
+                .map{ meta, _file ->
+                    def new_meta = meta + [process: "AUTOFILTER"]
+                    [new_meta, _file]
+                },
+            merged_table
+                .map{ meta, _file ->
+                    def new_meta = meta + [process: "MERGED_TABLE"]
+                    [new_meta, _file]
+                },
+            phylum_counts
+                .map{ meta, _file ->
+                    def new_meta = meta + [process: "MERGED_PHYLUM_COUNTS"]
+                    [new_meta, _file]
+                },
```

Comment:
To improve clarity - is it better to attach these process tags immediately after each process runs, rather than in here, as you do in other places?